packages/documentation/docs/SAG/SAG-for-starknet.md

```diff
@@ -1,6 +1,135 @@
+
+2. **Garaga for Multi-Scalar Multiplication (MSM)**: The implementation is using Garaga allowing efficient on-chain multi-scalar multiplication (MSM). The process incorporates hints passed alongside the ring signature to perform MSM efficiently.
+
+## Generating a Ring Signature for Starknet
+
+Here's how to generate a ring signature for starknet using Alice's Ring in TypeScript:
+
+```typescript
+import {
+  RingSignature,
+  Curve,
+  CurveName,
+  Point,
+} from "@cypher-laboratory/alicesring-sag-starknet";
+
+const curve = new Curve(CurveName.SECP256K1);
+const ring: Point[] = []; // Your ring of public keys
+const message = "Hello, Alice's Ring!";
+
+const signerPrivateKey = BigInt("0x..."); // The signer's private key
+async function ringSignForStarknet() {
+  // Sign
+  const signature = RingSignature.sign(ring, signerPrivateKey, message, curve);
+
+  // Verify locally
+  console.log("Is signature valid?", signature.verify()); // Should output: true
+
+  const signature = RingSignature.sign(ring, signerPrivKey, message, curve);
+  const rawCallData = await signature.getCallData();
+  console.log(rawCallData);
+}
+ringSignForStarknet(); 
+```
+
+**Important:** ringSignature.getCallData is an **async** function.
+
+## Verifying a Ring Signature On-Chain
+
+To verify the ring signature on-chain, you can use the [SAG verifier in Cairo](https://github.com/Cypher-Laboratory/Alice-s-Ring-Cairo-verifier).
+Below is an example of how to integrate the verifier into your Cairo contract.
+
+### Installation
+
+Add the SAG Cairo Verifier to your **Scarb.toml**:
+
+```toml
+[dependencies]
+alices_ring_cairo_verifier = {git = "https://github.com/Cypher-Laboratory/Alice-s-Ring-Cairo-verifier.git"}
+```
+
+### Importing the Verifier Interface
+
+```cairo
+use alices_ring_cairo_verifier::structtype::ringsignature;
+
+#[starknet::interface]
+pub trait IRingSignatureVerifier<TContractState> {
+    fn verify_rs(self: @TContractState, ring_signature: RingSignature) -> bool;
+}
+
+#[starknet::contract]
+mod RingSignatureVerifier {
+    use alices_ring_cairo_verifier::verify;
+    use alices_ring_cairo_verifier::structType::RingSignature;
+
+    #[storage]
+    struct Storage {}
+
+    #[abi(embed_v0)]
+    impl IRingSignatureVerifier of super::IRingSignatureVerifier<ContractState> {
+        fn verify_rs(self: @ContractState, ring_signature: RingSignature) -> bool {
+            verify(ring_signature)
+        }
+    }
+}
+```
+
+### Using Voyager block explorer :
+
+You can also pass the raw call data to the deployed contract at the following address:
+
+**Starknet Contract Address**: [`0x0317Ce745DE0A65308A19f6e184Be43A17F20D1eAe04B7433d984Df1D571DAA4`](https://sepolia.voyager.online/contract/0x0317Ce745DE0A65308A19f6e184Be43A17F20D1eAe04B7433d984Df1D571DAA4#readContract) on Sepolia Starknet.
+
```

Comment:
```suggestion
```

this is defined twice (see `Deployed address` section)

Reply:
this one refers to the action of making a call data to the block explorer (similar to this : https://garaga.gitbook.io/garaga/using-garaga-libraries-in-your-cairo-project/ec-multi-scalar-multiplication)

Reply:
can you explain this in the doc ?

Reply:
yep i will do a small video like in the garaga book